src/driver/i2c.c

```diff
@@ -16,28 +16,41 @@
 
 pthread_mutex_t i2c_mutex;
 
-void iic_init() {
-    pthread_mutex_init(&i2c_mutex, NULL);
-}
+#define IIC_PORTS 4
 
-static char *iic_ports[IIC_PORTS] = {
+static char *IIC_DEVS[IIC_PORTS] = {
     "/dev/i2c-0",
     "/dev/i2c-1",
     "/dev/i2c-2",
     "/dev/i2c-3",
 };
 
-static int iic_open(char *port) {
-    // int i2c_fd = open ("/dev/i2c-0", O_RDONLY);
-    int i2c_fd = open(port, O_RDONLY);
-    if (i2c_fd < 0) {
-        LOGE("iic_open %s failed [1=Right,2=Main,3=Left]", port);
+int g_iic_fds[IIC_PORTS];
+static int g_iic_report_once[IIC_PORTS] = {0};
+
+int iic_is_port_ready(int port) {
+    if (port < 0 || port >= IIC_PORTS) {
+        LOGE("Port %d contains an invalid range [0=N/A,1=Right,2=Main,3=Left]", port);
+        return 0;
+    } else if (g_iic_fds[port] < 0) {
+        if (!g_iic_report_once[port]) {
+            g_iic_report_once[port] = 1;
+            LOGE("Device %d:%s is not available [0=N/A,1=Right,2=Main,3=Left]", port, IIC_DEVS[port]);
+        }
+        return 0;
     }
-    return i2c_fd;
+    return 1;
 }
 
-static void iic_close(int i2c_fd) {
-    close(i2c_fd);
+void iic_init() {
+    pthread_mutex_init(&i2c_mutex, NULL);
+    int i;
+
+    // Offset starts with 1 as it is not referenced thus far.
+    for (i = 1; i < IIC_PORTS; ++i) {
```

Comment:
not a ton of reasons to pre-declare variables on modern compilers. lets please move iterator into the for statement.